src/ToastStack/ToastStack.tsx

```diff
@@ -0,0 +1,111 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import {CSSTransitionGroup} from "react-transition-group";
+import {ToastNotification} from "./ToastNotification";
+
+import {actionPropType} from "./ToastNotification";
+import {ToastType} from "./ToastType";
+
+import "./ToastStack.less";
+
+const propTypes = {
+  className: PropTypes.string,
+  defaultNotificationDurationMS: PropTypes.number,
+  notificationClassName: PropTypes.string,
+  notifications: PropTypes.arrayOf(PropTypes.shape({
+    action: actionPropType,
+    content: PropTypes.node.isRequired,
+    durationMS: PropTypes.number,
+    id: PropTypes.number.isRequired,
+    showCloseButton: PropTypes.bool,
+    type: PropTypes.oneOf(Object.values(ToastType)).isRequired,
+  })),
+  setNotifications: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  defaultNotificationDurationMS: 5000,
+};
+
+const cssClass = {
+  CONTAINER: "ToastStack",
+  NOTIFICATION_WRAPPER: "ToastStack--notificationWrapper",
+  ANIMATION: "ToastStack--animation",
+};
+
+/**
+ * Toast notifications can be used to provide feedback in response to user interaction. The
+ * <ToastStack> renders toast notifications and handles their enter/leave animations.
+ */
+export class ToastStack extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  _finiteDuration(notification) {
+    const {defaultNotificationDurationMS} = this.props;
+
+    if (notification.durationMS) {
+      return notification.durationMS !== Infinity;
+    }
+    return defaultNotificationDurationMS !== Infinity;
+  }
+
+  _clearNotification(id) {
+    const {notifications, setNotifications} = this.props;
+
+    setNotifications(notifications.filter(n => n.id !== id));
+  }
```

Comment:
I decided to have the `<ToastStack>` take in a `setNotifications` prop instead of a `clearNotification` prop for a few reasons:

1. This makes the `<ToastStack>`'s interface similar to other Dewey components. `<TextInput>` for example takes a `value` prop and an `onChange` prop to update `value`.
2. Constructing a `setNotifications` prop as a consumer is easier than constructing a `clearNotification` prop as a consumer. In fact, I initially planned to have this component take a `clearNotification` prop and constructed a buggy one when creating the `<ToastStack>` docs. Happy to describe the bug in person! (It involved outdated object references.)

Reply:
^ Actually, after trying to use this new `<ToastStack>` in an actual Clever repo, I've had a change of heart. Switched back from `setNotifications` to `clearNotification` in https://github.com/Clever/components/pull/350/commits/aec5da33fdc2d738651276b09bef5e57504eae97.

Regarding the reasoning for this:

1. Even with this new component, we'll still need to set up `clearNotification` Redux actions in our repos. Supporting a `setNotifications` API would require us to set up an additional, somewhat redundant Redux action.
2. The bugs that I'm concerned about are avoided in the Redux contexts that our `<ToastStack>`s will be used in.

I still went ahead and added a word of caution to the prop docs

Reply:
Left some initial notes about it (short of an actual explanation) in response to another comment. Let's chat tomorrow!